pkgs/applications/editors/pulsar/default.nix

```diff
@@ -0,0 +1,180 @@
+      hash = "sha256-gajIHUJrjCogcJjJVSKz85/x9inuS9rEMYlLqcolUqg=";
+    };
+  }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  additionalLibs = lib.makeLibraryPath [
+    xorg.libxshmfence
+    libxkbcommon
+    xorg.libxkbfile
+  ];
+  newLibpath = "${atomEnv.libPath}:${additionalLibs}";
+
+  # Hunspell for x86_64-linux
+  buildLocalePath = path: "searchPaths.push('${path}');";
+  localeDerivations = builtins.map (lang: hunspellDicts.${lang}) languages;
+  localePatchs = lib.concatMapStringsSep "" buildLocalePath localeDerivations;
+
+  # Hunspell for aarch64-linux
+  hunspellDirs = builtins.map (lang: "${hunspellDicts.${lang}}/share/hunspell") languages;
+  hunspellTargetDirs = "$out/opt/Pulsar/resources/app.asar.unpacked/node_modules/spellchecker/vendor/hunspell_dictionaries";
+  hunspellCopyCommands = lib.concatMapStringsSep "\n" (lang: "cp -r ${lang}/* ${hunspellTargetDirs};") hunspellDirs;
+in
+stdenv.mkDerivation rec {
+  inherit pname version;
+
+  src = with sourcesPath; fetchzip {
+    url = "https://github.com/${owner}/${pname}/releases/download/v${version}/${tarname}";
+    inherit hash;
+  };
+
+  nativeBuildInputs = [
+    wrapGAppsHook # Fix error: GLib-GIO-ERROR **: No GSettings schemas are installed on the system
+  ] ++ lib.optional useHunspell [
+    nodePackages.asar # We need ASAR to unpack the app.asar file and patch paths
+  ];
+
+  buildInputs = [
+    gtk3 # Fix error: GLib-GIO-ERROR **: Settings schema 'org.gtk.Settings.FileChooser' is not installed
+    xorg.libxkbfile # Fix error on electron: libxkbfile.so.1: cannot open shared object file: No such file or directory
+  ];
+
+  dontBuild = true;
+  dontConfigure = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/opt/Pulsar
+    mv * $out/opt/Pulsar
+
+    runHook postInstall
+  '';
+
+  preFixup = ''
+    gappsWrapperArgs+=(
+      # needed for gio executable to be able to delete files
+      --prefix "PATH" : "${lib.makeBinPath [ glib ]}"
+    )
+  '' + lib.optionalString (useHunspell && stdenv.hostPlatform.system == "x86_64-linux") ''
+    opt=$out/opt/Pulsar
+    # We need to patch the already existing app.asar.unpacked, and add python3
+    # to resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/node_gyp_bins/python3
+    rm $opt/resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/node_gyp_bins/python3
+    ln -s ${python3}/bin/python3 $opt/resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/node_gyp_bins/python3
+
+    # We need to extract the app.asar file
+    asar extract $opt/resources/app.asar ./app
+
+    # Remove lines 114 to 116 (inclusive) and replace with ''${localePaths}
+    sed -i '114,116d' ./app/node_modules/spell-check/lib/locale-checker.js
+    echo 'sed -i "114i ${localePatchs}" ./app/node_modules/spell-check/lib/locale-checker.js'
+    sed -i "114i ${localePatchs}" ./app/node_modules/spell-check/lib/locale-checker.js
```

Comment:
It should be possible if we create the patch at build-time (considering the languages to add is defined by `languages` in the argument attrSet. Is there a clean and official way to generate patchs at runtime ? (since providing `patches` to `mkDerivations`, as far as I can understand, can only take patches provided with this derivation)